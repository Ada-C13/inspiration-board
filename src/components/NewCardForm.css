/* The Modal (background) */

.new-card-form {
  width: 100%;
  padding-bottom: 4rem;
}

.new-card-form__header {
  text-align: center;
  text-transform: uppercase;
}

.new-card-form__form {
  width: 50%;
  margin: auto;
  font-size: 1.5em;
  display: grid;
  grid-template-columns: [labels] auto [controls] 1fr;
  grid-auto-flow: row;
}

.new-card-form__form-label  {
  grid-column: labels;
  grid-row: auto;
}

.new-card-form__form-select,
.new-card-form__form-textarea,
.new-card-form__form-button {
  font-family: 'Raleway', sans-serif;
  font-size: 1.25em;
  grid-column: controls;
  grid-row: auto;
  margin-bottom: .8em;
}

.new-card-form__form-button {
  background-color: inherit;
  border: 1px solid black;
  font-size: 1em;
}
