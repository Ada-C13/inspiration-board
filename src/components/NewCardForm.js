import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';
import './NewCardForm.css';

const EMOJI_LIST = ["", "heart_eyes", "beer", "clap", "sparkling_heart", "heart_eyes_cat", "dog"]

const NewCardForm =(props) => {

  const emojiDropdown = () => {
    const emojiList = EMOJI_LIST.map(emoji=>{
      return  (<option value={emoji}>{emoji}</option>);
    });
    return emojiList;
  };



  return(

    <form className="NewCardForm" onSubmit={props.onSubmitCard}>
    <label>Card Text:</label>
     <input name = "cardtext" type="text" onChange={props.onChangeCard}/>
    <label>Emoji Input:</label>
     <select  name="cardemoji" onChange={props.onChangeCard}>
      {emojiDropdown()}
    </select>
    <input
        type="submit"
        value="Submit Card"
      />
    </form>
  );
};

export default NewCardForm;