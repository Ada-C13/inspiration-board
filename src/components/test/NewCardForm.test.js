import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react';
import NewCardForm from '../NewCardForm';

describe('NewCardForm', () => {
  afterEach(cleanup);
  
  test('that it matches the existing snapshot', () => {
    // Arrange-Act 
    const { asFragment } = render(
      <NewCardForm 
        addCardCallback={() => { }}
      />
    );

    // Assert 
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });


  describe('User Interaction', () => {
    // render the form and get the inputs & button
    const setup = () => {
      const addCardCallbackMock = jest.fn(() => { });
      const renderResult = render(<NewCardForm addCardCallback={addCardCallbackMock}/>);
      const textInput = renderResult.getByTestId('new-card-form--text');
      const emojiInput = renderResult.getByTestId('new-card-form--emoji'); 
      const addButton = renderResult.getByText('Add Card');
      const form = renderResult.getByTestId('new-card-form--form');

      return {
        ...renderResult,
        addCardCallbackMock,
        textInput,
        emojiInput,
        addButton,
        // form
      }
    };

    test('Typing on the text field changes the input', () => {
      // Arrange 
      const formData = setup();
      const simulatedEvent = {
        target: {
          value: 'howdy world',
        },
      };

      // Act 
      // put 'howdy world' in the input field 
      fireEvent.change(formData.textInput, simulatedEvent);

      // Assert 
      expect(formData.textInput.value).toBe('howdy world');
    });

    test('Selecting on the emoji field changes the input', () => {
      // Arrange 
      const formData = setup();
      const simulatedEvent = {
        target: {
          value: "heart_eyes",
        },
      };

      // Act 
      // put 'heart_eyes' in the input field
      fireEvent.change(formData.emojiInput, simulatedEvent);

      // Assert 
      expect(formData.emojiInput.value).toBe('heart_eyes');
    });


    test('The callback function is called correctly when the form is submitted', () => {
      // arrange
      const formData = setup();

      // Act 
      fireEvent.change(formData.textInput, { target: { value: 'howdy world', }, })
      fireEvent.change(formData.emojiInput, { target: { value: 'heart_eyes', }, })
      fireEvent.click(formData.addButton);

      expect(formData.addCardCallbackMock).toHaveBeenCalled();
      expect(formData.textInput.value).toBe('');
      expect(formData.emojiInput.value).toBe('');
    });
  });
});