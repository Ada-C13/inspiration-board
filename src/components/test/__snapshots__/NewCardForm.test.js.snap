// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewCardForm that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <h2
    class="new-card-form__header"
  >
    Add a new card!!
  </h2>
  <div
    class="new-card-form"
  >
    <form
      class="new-card-form__form"
      data-testid="new-card-form--form"
    >
      <div>
        <label
          class="new-card-form__form-label"
          for="text"
        />
        <textarea
          class="new-card-form__form-textarea"
          data-testid="new-card-form--text"
          id="text"
          name="text"
          placeholder="Type something..."
        />
      </div>
      <div>
        <label
          class="new-card-form__form-label"
          for="emoji"
        />
        <select
          class="new-card-form__form-select"
          data-testid="new-card-form--emoji"
          id="emoji"
          name="emoji"
        >
          <option
            value=""
          >
             --emoji-- 
          </option>
          <option
            value="heart_eyes"
          >
            😍
          </option>
          <option
            value="beer"
          >
            🍺
          </option>
          <option
            value="clap"
          >
            👏
          </option>
          <option
            value="sparkling_heart"
          >
            💖
          </option>
          <option
            value="heart_eyes_cat"
          >
            😻
          </option>
          <option
            value="dog"
          >
            🐶
          </option>
        </select>
        <input
          class="new-card-form__form-button"
          type="submit"
          value="Add Card"
        />
      </div>
    </form>
  </div>
</DocumentFragment>
`;
